modules/edu.gemini.seqexec.web/edu.gemini.seqexec.web.client/src/main/scala/edu/gemini/seqexec/web/client/model/handlers.scala

```diff
@@ -492,6 +493,20 @@ object handlers {
         updated(value.copy(user = u))
     }
 
+    val stepCompletedMessage: PartialFunction[Any, ActionResult[M]] = {
+      case ServerMessage(e @ StepExecuted(obsId, sv)) =>
+      val curStep =
+        for {
+            obs      <- sequenceViewT.find(_.id === obsId)(e)
+            curSIdx  <- obs.runningStep.map(_._1)
+            curStep  <- sequenceStepT.find(_.id === curSIdx)(obs)
+            if curStep.observeStatus === ActionStatus.Pending && curStep.status === StepState.Running
```

Comment:
@jluhrs This is odd to me. I'd expect to check for `curStep.observeStatus === ActionStatus.Completed` but the state from the server at the end of the observation is `Pending`. Any thoughts?

Reply:
I think there is a bug.